{
  try {
    doStop();
    container.stop();
  }
 catch (  Exception e) {
  }
  return this;
}
